tests/jballerina-unit-test/src/test/resources/test-src/expressions/varref/record-variable-reference.bal

```diff
@@ -191,56 +191,205 @@ function testRestParameterType() returns boolean {
     map<anydata|error> other1 = {};
     map<any> other2 = {};
 
-    IntRestRecord rec1 = { name: "A", married: true, "age": 19, "token": 200 };
-    { name, ...other1 } = rec1;
+    IntRestRecord rec1 = {name: "A", married: true, "age": 19, "token": 200};
+    {name, ...other1} = rec1;
 
     any a1 = other1;
 
     return a1 is map<anydata|error>;
 }
 
-// TODO: Uncomment below tests once record literal is supported with var ref
+function testVarAssignmentOfRecordLiteral1() {
+    string fName;
+    boolean married;
+    int theAge;
+    string format;
 
-//function testVarAssignmentOfRecordLiteral() returns (string, boolean, int, string) {
-//    string fName;
-//    boolean married;
-//    int theAge;
-//    string format;
-//    map theMap;
-//
-//    {name: fName, age: {age: theAge, format}, married, ...theMap} = {name: "Peter", married: true, age: {age:12, format: "Y"}};
-//    return (fName, married, theAge, format);
-//}
-//
-//function testVarAssignmentOfRecordLiteral2() returns (string, boolean, Age) {
-//    string fName;
-//    boolean married;
-//    Age age;
-//    map theMap;
-//
-//    {name: fName, age, married, ...theMap} = {name: "Peter", married: true, age: {age:12, format: "Y"}};
-//    return (fName, married, age);
-//}
-//
-//function testVarAssignmentOfRecordLiteral3() returns (string, boolean, map) {
-//    string fName;
-//    boolean married;
-//    map age;
-//    map theMap;
-//
-//    {name: fName, age, married, ...theMap} = {name: "Peter", married: true, age: {age:12, format: "Y"}};
-//    return (fName, married, age);
-//}
-//
-//function testVarAssignmentOfRecordLiteral4() returns (string, boolean, json) {
-//    string fName;
-//    boolean married;
-//    json age;
-//    map theMap;
-//
-//    {name: fName, age, married, ...theMap} = {name: "Peter", married: true, age: {age:12, format: "Y"}};
-//    return (fName, married, age);
-//}
+    {
+        name: fName,
+        age: {
+            age: theAge,
+            format
+        },
+        married
+    } = {
+        name: "Peter",
+        married: true,
+        age: {age: 12, format: "Y"}
+    };
+
+    assertEquality("Peter", fName);
+    assertEquality(true, married);
+    assertEquality(12, theAge);
+    assertEquality("Y", format);
+}
+
+function testVarAssignmentOfRecordLiteral2() {
+    string fName;
+    boolean married;
+    Age age;
+    map<anydata> theMap;
+
+    {
+        name: fName,
+        age,
+        married,
+        ...theMap
```

Comment:
theMap isn't asserted

Reply:
That's intentional. We don't need to assert it. We have asserted it in the other tests.

Reply:
we can remove this test. It seems this test is not necessary.

Reply:
Fixed via [6568a25](https://github.com/ballerina-platform/ballerina-lang/pull/40283/commits/6568a25b943625b81aba3d44930295716bca906b)